templates/seL4VMParameters.template.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2023, DornerWorks
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <camkes.h>
+#include <arm_vm/gen_config.h>
+
+/*- set vm_address_config = configuration[me.name].get('vm_address_config') -*/
+/*- set vm_image_config = configuration[me.name].get('vm_image_config') -*/
+/*- set linux_address_config = configuration[me.name].get('linux_address_config') -*/
+/*- set linux_image_config = configuration[me.name].get('linux_image_config') -*/
+
+/*- if vm_address_config -*/
+
+unsigned long ram_base = /*? vm_address_config.get('ram_base') ?*/;
+unsigned long ram_paddr_base = /*? vm_address_config.get('ram_paddr_base') ?*/;
+unsigned long ram_size = /*? vm_address_config.get('ram_size') ?*/;
+unsigned long ram_offset = /*? vm_address_config.get('ram_offset') ?*/;
```

Comment:
Is this used anywhere?

Reply:
I'm not sure. If its unused, it should be removed as part of a new PR.

Reply:
I can't find a reference. Since we are creating a new block here, I'd leave everything out that has no clear use case, as there seem no gain just copying legacy. So, does anybody know where this is used?

Reply:
Again, I'm not opposed to removing it, it just feels like that is a feature that deserves its own PR as its not related to the overall topic of this PR.

Reply:
True, it's just that we've moved now from changing name to creating a new config block. And we keep the old one for compatibility for a while. So, for the new block we can start small and just pass the test and use cases we have. If somebody runs into an issue during migration later, we can always extend things easily, it's a small PR and then also a clear pointer where this is used. It's much harder to remove something that has a blurry usage already.

Reply:
@kent-mcleod do you think that removing the initrd_max_size and ram_offset should be a part of this commit?

Reply:
ram_offset's dependency was removed in https://github.com/seL4/camkes-vm-examples/commit/da44d5b3c6ca9dece94163dc170334cef1ebabf3.  It seems like that would indicate that ram_offset was only ever set to 0.  I can't find where initrd_max_size was ever used. Now that we've looked it up, it's probably worth removing them from the new config variation.

Reply:
Alright, I've removed them from the new config variation, and I also remove some extra macros that were in the vmlinux.h files.